### PR TITLE
HW 17. Async/await, enum, generic using unit tests

### PR DESCRIPTION
**Pull request by yustasstar**

### CODE REVIEW

File: NUnitTests/Lessons/21/EnumHomework.cs [hunk 1]
```diff
@@ -1,42 +1,41 @@
...
     }
 
     [Test]
     public void SomeIntCorrespondsToSomeTestDataAgeValue()
     {
         var listOfInt = new List<int>() { 5, 14, 15 };
-
-        var isAnyIntCorrespondsToTestDataAge = 
-
+        var isAnyIntCorrespondsToTestDataAge = listOfInt.Any(x => x == (int)TestDataAge.Teenager);
```
Comment (vldemyan) [inline NUnitTests/Lessons/21/EnumHomework.cs]: we are interested to know if some int from listOfInt is equal to some int from TestDataAge. "some" means any of  Child,     Teenager, Adult, but not only to Teenager

File: NUnitTests/Lessons/21/EnumHomework.cs [hunk 2]
```diff
@@ -46,16 +45,24 @@ public void NumberOfIntCorrespondsToSomeTestDataAgeValue()
...
-        var numberOfStringsWhichPresentInEnum = 
-        var numberOfStringsWhichAreNotPresentInEnum = 
-        var areAllPresent = 
-        var areExtraElements = 
+        // Calculate the number of strings present in the enum
+        var numberOfStringsWhichPresentInEnum = listOfString.Count(str => enumValues.Contains(str));
+        // Calculate the number of strings not present in the enum
+        var numberOfStringsWhichAreNotPresentInEnum = listOfString.Count(str => !enumValues.Contains(str));
+        // Determine if all strings are present in the enum
+        var areAllPresent = numberOfStringsWhichAreNotPresentInEnum == 0;
```
Comment (vldemyan) [inline NUnitTests/Lessons/21/EnumHomework.cs]: this is LINQ task. Please do this according to description "Use such LINQ as Any(), Count(), Contains()"

File: NUnitTests/Lessons/21/EnumHomework.cs [hunk 3]
```diff
@@ -46,16 +45,24 @@ public void NumberOfIntCorrespondsToSomeTestDataAgeValue()
...
+        // Determine if there are extra elements not present in the enum
+        var areExtraElements = numberOfStringsWhichAreNotPresentInEnum > 0;
```
Comment (vldemyan) [inline NUnitTests/Lessons/21/EnumHomework.cs]: the same

File: NUnitTests/Lessons/21/EnumHomework.cs (hunk 2 above)
Comment (yustasstar) [reply NUnitTests/Lessons/21/EnumHomework.cs]: fixed please check

File: NUnitTests/Lessons/21/EnumHomework.cs (hunk 3 above)
Comment (yustasstar) [reply NUnitTests/Lessons/21/EnumHomework.cs]: fixed please check

File: NUnitTests/Lessons/21/EnumHomework.cs (hunk 1 above)
Comment (yustasstar) [reply NUnitTests/Lessons/21/EnumHomework.cs]: fixed please check

Comment (vldemyan) [reply NUnitTests/Lessons/21/EnumHomework.cs]: thanks, closed

File: NUnitTests/Lessons/21/EnumHomework.cs (hunk 2 above)
Comment (vldemyan) [reply NUnitTests/Lessons/21/EnumHomework.cs]: thanks, closed

File: NUnitTests/Lessons/21/EnumHomework.cs (hunk 3 above)
Comment (vldemyan) [reply NUnitTests/Lessons/21/EnumHomework.cs]: thanks, closed

**Review by vldemyan — APPROVED** [review]

**Pull request merged**